cachi2/core/resolver.py

```diff
@@ -0,0 +1,30 @@
+from typing import Callable
+
+from cachi2.core.errors import UnsupportedFeature
+from cachi2.core.models.input import PackageManagerType, Request
+from cachi2.core.models.output import RequestOutput, merge_outputs
+from cachi2.core.package_managers import gomod
+
+Handler = Callable[[Request], RequestOutput]
+
+_package_managers: dict[PackageManagerType, Handler] = {
+    "gomod": gomod.fetch_gomod_source,
+}
+
+
+supported_package_managers = list(_package_managers)
+
+
+def resolve_packages(request: Request) -> RequestOutput:
+    """Run all requested package managers, return their combined output."""
+    # de-dupe but preserve order
+    requested_types = dict.fromkeys(pkg.type for pkg in request.packages).keys()
```

Comment:
I was trying to figure out why we need to dedup and preserve order here, but I couldn't. :sweat_smile: 

Can we use a set here instead?

Reply:
I figured if the user specifies `--package pip --package gomod`, they might expect pip packages to be processed first. But `--package pip --package gomod --package pip` would end up breaking their expectations and going pip -> pip -> gomod anyway, so might as well use a `sorted(set())`

Reply:
Changed to a set